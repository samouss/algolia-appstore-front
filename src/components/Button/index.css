.Button {
  border: 1px solid #0080ff;
  border-radius: 3px;
  background-color: #ffffff;
  padding: 5px 8px;
  color: #0080ff;
  text-transform: uppercase;
  white-space: nowrap;
  text-decoration: none;
  font-weight: 500;
  font-size: 0.9rem;

  @media (--tablet) {
    font-size: 1.1rem;
  }
}
