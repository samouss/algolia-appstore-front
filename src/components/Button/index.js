import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Button = ({ to, children, ...props }) => (
  <a
    {...props}
    href={to}
    styleName="Button"
  >
    {children}
  </a>
);

Button.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
