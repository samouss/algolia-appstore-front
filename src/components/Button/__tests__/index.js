import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';

describe('<Button />', () => {
  const defaultProps = {
    to: 'http://link.com',
    target: '_blank',
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Button
        {...props}
      >
        Link
      </Button>,
    );

    expect(component).toMatchSnapshot();
  });
});
