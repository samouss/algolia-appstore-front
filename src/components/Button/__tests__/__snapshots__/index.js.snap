// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> expect to render 1`] = `
<a
  href="http://link.com"
  styleName="Button"
  target="_blank"
>
  Link
</a>
`;
